cmd/snap/common.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	sys "syscall"
+	"time"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/i18n"
+	"github.com/ubuntu-core/snappy/lockfile"
+	"github.com/ubuntu-core/snappy/snappy"
+)
+
+func isAutoUpdateRunning() bool {
+	unitName := "snappy-autopilot"
+	bs, err := exec.Command("systemctl", "show", "--property=SubState", unitName).CombinedOutput()
+	if err != nil {
+		return false
+	}
+
+	return strings.TrimSpace(string(bs)) == "SubState=running"
+}
+
+// withMutexAndRetry runs the given function with a filelock mutex and provides
+// automatic re-try and helpful messages if the lock is already taken
```

Comment:
// FIXME: This function is a left over from the old CLI and a major red flag. No client-side locks should exist.